theano/sandbox/gpuarray/opt.py

```diff
@@ -254,10 +255,43 @@ def local_gpu_elemwise(node):
     name = op.name
     if name:
         name = 'Gpu'+name
+
     res = GpuElemwise(scal_op, name=name,
-                      inplace_pattern=copy.copy(op.inplace_pattern),
-                      nfunc_spec=op.nfunc_spec)
-    return res
+                          inplace_pattern=copy.copy(op.inplace_pattern),
+                          nfunc_spec=op.nfunc_spec)
+
+    # If the elemwise operation is a pow, casts might be required on the
+    # inputs and or outputs because only the (float, float)->float and
+    # (double, double)->double cases are implemented at the moment.
+    if isinstance(op.scalar_op, Pow):
+        old_out_dtype = node.outputs[0].dtype
+        old_inp_dtypes = [inp.dtype for inp in node.inputs]
+        new_out_dtype = upcast("float32", *old_inp_dtypes)
```

Comment:
The upcast was meant to ensure that the resulting dtype would be floating point and not an integer (which is not currently supported). Can't I simply upcast with 'float16' instead of 'float32'?